lib/Jsrt/JsrtHelper.cpp

```diff
@@ -89,40 +91,32 @@ void JsrtCallbackState::ObjectBeforeCallectCallbackWrapper(JsObjectBeforeCollect
     }
 
 #ifndef _WIN32
-    static pthread_key_t s_threadLocalDummy;
+    static THREAD_LOCAL struct _OnThreadExit
+    {
+        ~_OnThreadExit()
+        {
+            ThreadBoundThreadContextManager::DestroyContextAndEntryForCurrentThread();
+        }
+    } _s_onThreadExit;
 #endif
-    static THREAD_LOCAL bool s_threadWasEntered = false;
 
-    _NOINLINE void DISPOSE_CHAKRA_CORE_THREAD(void *_)
-    {
-        free(_);
-        ThreadBoundThreadContextManager::DestroyContextAndEntryForCurrentThread();
-    }
+    static THREAD_LOCAL bool s_threadWasEntered = false;
 
     _NOINLINE bool InitializeProcess()
     {
-#if !defined(_WIN32)
-        pthread_key_create(&s_threadLocalDummy, DISPOSE_CHAKRA_CORE_THREAD);
-#endif
-
-    // setup the cleanup
-    // we do not track the main thread. When it exits do the cleanup below
+        // setup the cleanup
+        // we do not track the main thread. When it exits do the cleanup below
 #ifdef CHAKRA_STATIC_LIBRARY
-    atexit([]() {
-        ThreadContext *threadContext = ThreadContext::GetContextForCurrentThread();
-        if (threadContext)
-        {
-            if (threadContext->IsInScript()) return;
-            ThreadBoundThreadContextManager::DestroyContextAndEntryForCurrentThread();
-        }
+        atexit([]() {
+            ThreadBoundThreadContextManager::DestroyAllContextsAndEntries();
```

Comment:
It looks the added THREAD_LOCAL destructor is called before this which handled it already.